src/FormatCommand.cs

```diff
@@ -58,11 +62,84 @@ internal static RootCommand CreateCommandLineOptions()
             return rootCommand;
         }
 
-        internal static bool WasOptionUsed(this ParseResult result, params string[] aliases)
+        private static LogLevel? ParseLogLevel(ArgumentResult result)
+        {
+            if (result.Tokens.Count > 1)
+            {
+                result.ErrorMessage = "Cannot specify verbosity more than once.";
+                return null;
+            }
+
+            if (result.Tokens.Count == 0)
+            {
+                result.ErrorMessage = "Must specify a verbostiy level.";
+                return null;
+            }
+
+            var verbosity = result.Tokens.Single();
+            return verbosity.Value.ToLowerInvariant() switch
+            {
+                "q" => LogLevel.Error,
+                "quiet" => LogLevel.Error,
+                "m" => LogLevel.Warning,
+                "minimal" => LogLevel.Warning,
+                "n" => LogLevel.Information,
+                "normal" => LogLevel.Information,
+                "d" => LogLevel.Debug,
+                "diag" => LogLevel.Debug,
+                "diagnostic" => LogLevel.Trace,
+                _ => LogLevel.Information
+            };
+        }
+
+        private static DiagnosticSeverity? ParseDiagnosticSeverity(ArgumentResult result)
         {
-            return result.Tokens
-                .Where(token => token.Type == TokenType.Option)
-                .Any(token => aliases.Contains(token.Value));
+            if (result.Tokens.Count > 1)
+            {
+                result.ErrorMessage = $"Cannot specify severity more than once for {result.Argument.Name}";
+                return null;
+            }
+
+            if (result.Tokens.Count == 0)
+            {
+                // Use Defaults
+                return DiagnosticSeverity.Error;
+            }
+
+            var severityValue = result.Tokens.Single().Value;
+            if (!HasMatch(severityValue, out var severity))
+            {
+                // Assume that this means our arity is zero and the parser should re-parse accordingly
+                result.ErrorMessage = $"Cannot specify severity more than once for {result.Argument.Aliases.FirstOrDefault()}";
```

Comment:
Error message not applicable

Reply:
Is this possible since we mark the argument with valid values using FromAmong?

Reply:
blocked by https://github.com/dotnet/command-line-api/pull/955